CleverTap/Runtime/Native/UnityNativeWrapper/Models/UnityNativeAppInboxPersistence.cs

```diff
@@ -93,7 +93,15 @@ public static void MarkAsRead(string id)
                 return;
             }
 
-            _preferenceManager.SetInt($"{_readPrefixKey}_{id}", 1);
+            id = $"{_readPrefixKey}_{id}";
+
+            // Updating the message to mark it as read
+            string msg = _preferenceManager.GetString(id, string.Empty);
+            Dictionary<string, object> message = Json.Deserialize(msg) as Dictionary<string, object>;
```

Comment:
If the `msg` is the default value `string.Empty`, the deserialization and the rest of the method will fail.